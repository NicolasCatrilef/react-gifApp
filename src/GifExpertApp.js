import React, { useState } from "react";
import { AddCategory } from "./component/AddCategory";
import { GifGrid } from "./component/GifGrid";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState([ 'Kimetsu No Yaiba' ]);

    // const handleAdd = ( e ) => {

    //     setCategories( [ 'Kimetsu No Yaiba', ...categories ] );

    // }
    
    return (
        <>
            <h2> Gif Expert App </h2>
            <AddCategory setCategories={ setCategories }/>
            <hr/>

            {/* <button onClick = { handleAdd }> Agregar </button> */}

            <ol>
                {
                    categories.map( category => (
                        <GifGrid 
                            key = { category }
                            category = { category } 
                        />
                    ) )
                }
            </ol>
        </>
    );
}