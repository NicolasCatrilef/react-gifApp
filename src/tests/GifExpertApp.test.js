import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";


describe('Pruebas GifExpertApp', () => {
    
    test('Renderizar GifExpertApp', () => {
        
        const wrapper = shallow( <GifExpertApp /> );

        expect( wrapper ).toMatchSnapshot();
    });
    
    test('Cagi', () => {
        const cateogries = [ 'Kimetsu No Yaiba', 'Darling in the Franxx' ];
        const wrapper = shallow( <GifExpertApp defaultCategories = { cateogries }/> );
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( 'GifGrid' ).length ).toBe( cateogries.length );  
    })
    
});
