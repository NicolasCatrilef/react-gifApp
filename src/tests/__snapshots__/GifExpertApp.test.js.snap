// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GifExpertApp Cagi 1`] = `
<Fragment>
  <h2>
     Gif Expert App 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Kimetsu No Yaiba"
      key="Kimetsu No Yaiba"
    />
    <GifGrid
      category="Darling in the Franxx"
      key="Darling in the Franxx"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas GifExpertApp Renderizar GifExpertApp 1`] = `
<Fragment>
  <h2>
     Gif Expert App 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
